### PR TITLE
Enable more clang-tidy checks.

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
# Issue

It is continuation of https://github.com/Project-OSRM/osrm-backend/pull/6261

## Tasklist

 - [x] CHANGELOG.md entry ([How to write a changelog entry](http://keepachangelog.com/en/1.0.0/#how))
 - [ ] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki)
 - [ ] add tests (see [testing documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/testing.md))
 - [ ] review
 - [ ] adjust for comments
 - [ ] cherry pick to release branch

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

**Comment by SiarheiFedartsou** [comment]
@mjjbell please take a look when you have a chance.

File: src/guidance/turn_handler.cpp [hunk 1]
```diff
@@ -684,7 +684,7 @@ bool TurnHandler::isCompatibleByRoadClass(const Intersection &intersection, cons
 boost::optional<TurnHandler::Fork> TurnHandler::findFork(const EdgeID via_edge,
                                                          Intersection &intersection) const
 {
-    const auto fork = findForkCandidatesByGeometry(intersection);
+    auto fork = findForkCandidatesByGeometry(intersection);
```
Comment (SiarheiFedartsou) [inline src/guidance/turn_handler.cpp:687]: Causes https://clang.llvm.org/extra/clang-tidy/checks/performance/no-automatic-move.html

**Comment by SiarheiFedartsou** [comment]
Something weird happens on CI:
```
 Cannot initiate the connection to ppa.launchpad.net:80 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 80]
```

**Comment by mjjbell** [comment]
You should be able to run `make tests` from your build directory if you a faster feedback loop.

**Comment by SiarheiFedartsou** [comment]
> You should be able to run `make tests` from your build directory if you want a faster feedback loop.

Sorry for that and thank you for this tip, tbh I didn't know how to build tests locally and decided that it would be easy to just wait for a couple of CI runs to fix all the issues, but it turned out there are too much of them.

**Review by mjjbell — APPROVED** [review]
Looks good, thanks 👍

**Pull request merged**